### PR TITLE
Use Test::Warnings where possible to prevent warnings in tests

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Comment by okurz** [comment]
@aaannz one warning that is catched by the above tests is
https://github.com/os-autoinst/os-autoinst/blob/master/needle.pm#L45
when `PRJDIR` is not defined. How should that code behave?

**Comment by aaannz** [comment]
Ah right! Didn't look into the https://progress.opensuse.org/issues/16362 yet. But will do in a way that PRJDIR is mandatory and CASEDIR, PRODUCTDIR etc., if not set, should be derived from PRJDIR. Can be?

**Comment by aaannz** [comment]
Hmm.. hold that thought. PRJDIR doesn't contain distri etc. so no.. better to infer PRJDIR from CASEDIR.

**Comment by okurz** [comment]
good. This PR is not urgent to be merged but I think it is helpful to use to fix the other issues.

File: cpanfile [hunk 1]
```diff
@@ -52,10 +52,12 @@ on 'test' => sub {
   requires 'Perl::Critic';
   requires 'Test::Output';
   requires 'Test::Fatal';
+  requires 'Test::Warnings';
   requires 'Test::Pod';
   requires 'Test::MockModule';
   requires 'Pod::Coverage';
   requires 'Devel::Cover';
+  requires 'String::Random';
```
Comment (coolo) [inline cpanfile]: this looks like a merge victim

**Comment by okurz** [comment]
updated

**Pull request closed without merging**